homeassistant/components/number/__init__.py

```diff
@@ -48,12 +48,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     component.async_register_entity_service(
         SERVICE_SET_VALUE,
         {vol.Required(ATTR_VALUE): vol.Coerce(float)},
-        "async_set_value",
+        async_set_value,
     )
 
     return True
 
 
+async def async_set_value(entry: NumberEntity, async_service_call: Callable) -> None:
```

Comment:
Its the service call instance that is being passed here.
```suggestion
async def async_set_value(entry: NumberEntity, service_call: ServiceCall) -> None:
```